### DISCUSSION TITLE
Are pthread includes broken?

### DISCUSSION THREAD

**Discussion by lolgear** [libgit2/libgit2#5975] [Q&A] [answered]
I guess for macOS import statements should have prefix "pthread"

```c
// Old
# ifdef GIT_THREADS
#	include <pthread.h>
#	include <sched.h>
# endif
```

```c
// New
# ifdef GIT_THREADS
#	include <pthread/pthread.h>
#	include <pthread/sched.h>
# endif
```

**Comment by ethomson** [comment]
> I guess for macOS import statements should have prefix "pthread"

Should they?  I build on macOS daily and have never faced this problem, plus this is a posix standard.  I feel like this should be unnecessary (and is if you use cmake).  What's the difference between how cmake invokes the compiler and how you're doing it that makes this change necessary?

**Reply by lolgear** [reply]
Swift PM treats them differently.
#5979 
If you remove this change, SPM will try to search in local `pthread.h` header.

**Comment by ethomson** [comment, marked as answer]
I assume that's because this line https://github.com/libgit2/libgit2/pull/5979/files#diff-f913940c58e8744a2af1c68b909bb6383e49007e6c5a12fb03104a9006ae677eR90 translates into a `-Isrc/unix`?

Is that really necessary?  We should always include `unix/pthread.h` _as_ `unix/pthread.h` so the `-Isrc` should suffice.

I don't want to `ifdef` here and hope that `pthread/pthread.h` always works on macOS.  The posix standard calls this `pthread.h`.  If we really need to distinguish the two, then we should rename `unix/pthread.h` to something else.  But even that shouldn't be necessary.

**Reply by lolgear** [reply]
Yeah, I opened an issue at Swift bug tracker. It seems that SPM "header search path" cSetting overrides default value and SPM can't find system headers.

**Reply by lolgear** [reply]
I fixed a bug with search paths.
